mainloop.cpp

```diff
@@ -384,12 +384,10 @@ bool Mainloop::add_endpoints(Mainloop &mainloop, struct opt *opt)
                 continue;
             }
 
-            if (_add_tcp_endpoint(tcp.get()) < 0) {
+            if (_add_tcp_endpoint(tcp.release()) < 0) {
```

Comment:
I'd say to leave the get() here and the release() below. Then everything is correct again.